src/main/java/uk/co/ramp/covid/simulation/place/Place.java

```diff
@@ -118,18 +131,26 @@ public void doInfect(Time t, DailyStats stats, ContactsWriter contactsWriter) {
        stepInfections(t, stats);
 
         for (Person cPers : people) {
-            if (cPers.isInfectious()) {
-                for (Person nPers : people) {
-                    if (cPers != nPers && !nPers.getInfectionStatus()) {
-                        double transP = getTransP(t, cPers, nPers);
-                        boolean infected = nPers.infChallenge(transP);
-                        if (infected) {
-                            registerInfection(t, nPers, stats);
-                            nPers.getcVirus().getInfectionLog().registerInfected(t);
-                            cPers.getcVirus().getInfectionLog().registerSecondaryInfection(t, nPers);
-                        }
-                    }
-                }
+            if (cPers.isInfectious() && getNumPeople() > 1) {
+            	int nInfected = rng.nextBinomial(getNumPeople() - 1, getTransP(cPers) / 24.0);
+            	List <Person> usedPerson = new ArrayList<>();
+            	if(nInfected > 0) {
+            		for(int nextInt = 1; nextInt <= nInfected; nextInt++) {
```

Comment:
Are we sure this always terminates? E.g. if everyone at a place is already infected?

Looks like we can also infect already infected people with the checks below (which I don't think we want? Unless these are skipped by infChallenge).

Reply:
I am sure that this can't get stuck in a loop because the arithmetic is on our side. However, the most recent commit has a veyr ugly get-out. Have a look please but we may decide that this is too unagainly

Reply:
Agree the getOut is a liter ugly. How much does it matter if we remove the final nextInt-- section? That is, we try to infect nInfected *if possible* and even though we might be a little short sometimes, most of the time we will get close?
That way we ensure we always terminate?

Happy for you to also say no if this seriously messes up the epidemiology, and we can leave in this get-out clause (but probably less than 1m retries, maybe numPeople * 10 or something is more appropriate?).

Reply:
Alright. We definitely need the nextInt--, to drop it would be a fairly intolerable assumption in places with few people. I ahve made it a funciton of getNumPeople()